src/git/git.go

```diff
@@ -232,23 +238,23 @@ func (g *GitVersioning) getMostRecentTag() (string, error) {
 		return "0.0.0", nil
 	}
 
-	mapTags := make(map[int]string)
+	mapTags := make(map[*Version]string)
 
 	for _, currentTag := range g.tagsList {
 		tag := strings.TrimSpace(strings.Replace(currentTag.Name, "refs/tags/", "", 1))
 
 		if pattern.MatchString(tag) {
-			tagInt, _ := strconv.Atoi(strings.ReplaceAll(tag, ".", ""))
-			mapTags[tagInt] = tag
+			version := newVersion(tag)
+			mapTags[version] = tag
 		}
 	}
 
-	result := ""
-	previous := 0
-	for key, element := range mapTags {
-		if key > previous {
-			previous = key
-			result = element
+	var latest *Version
+	var result string
+	for version, tag := range mapTags {
+		if latest == nil || version.isGreaterThan(latest) {
```

Comment:
maybe to improve code reading... create a function like this

    isSetNewVersion(latest, version)

@luc-zago

Reply:
@LeoCBS good suggestion. Done :heavy_check_mark: